From 0a85b888b0384e83fa4a80c3751a23f4de8a9310 Mon Sep 17 00:00:00 2001
From: Fabio M. Di Nitto <fdinitto@redhat.com>
Date: Tue, 21 Aug 2012 10:49:20 +0000
Subject: [PATCH] sa-confdb: fix crash due to incorrect malloc size

with recent changes, confdb and saconfdb were allocating two different
sizes for the same results. While on x86_64 this was masked nicely,
on i686 it causes crashes.

Signed-off-by: Fabio M. Di Nitto <fdinitto@redhat.com>
Reviewed-by: Jan Friesse <jfriesse@redhat.com>
---
 lib/sa-confdb.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/sa-confdb.c b/lib/sa-confdb.c
index 9a11418..7a8fabf 100644
--- a/lib/sa-confdb.c
+++ b/lib/sa-confdb.c
@@ -263,7 +263,7 @@ int confdb_sa_key_get_typed (
 				    &kvalue, value_len, (objdb_value_types_t*)type);
 	if (!res) {
 		if (!*value) {
-			*value = malloc(*value_len);
+			*value = malloc(*value_len + 1);
 			if (!*value) {
 				res = CS_ERR_NO_MEMORY;
 			}
@@ -443,7 +443,7 @@ int confdb_sa_key_iter_typed (
 		memcpy(key_name, kname, key_name_len);
 		key_name[key_name_len] = '\0';
 		if (!*value) {
-			*value = malloc(*value_len);
+			*value = malloc(*value_len + 1);
 			if (!*value) {
 				res = CS_ERR_NO_MEMORY;
 			}
-- 
1.7.1

